File: csharp/bob/Bob.cs
using System;
using System.Linq;

public static class Bob
{
    public static string Response(string statement)
    {
        statement = statement.Trim();
        return statement switch
        {
            _ when IsYellQuestion(statement) => "Calm down, I know what I'm doing!",
            _ when IsYell(statement) => "Whoa, chill out!",
            _ when IsQuestion(statement) => "Sure.",
            _ when IsIdleTalk(statement) => "Fine. Be that way!",
            _ => "Whatever."
        };
    }

    private static bool HasLetters(string str) => str.Any(char.IsLetter);

    private static bool IsQuestion(string str) => str.EndsWith("?");

    private static bool IsYell(string str) => HasLetters(str) && str.Equals(str.ToUpper());

    private static bool IsYellQuestion(string str) => IsQuestion(str) && IsYell(str);

    private static bool IsIdleTalk(string str) => string.IsNullOrWhiteSpace(str);
}

File: go/bob/bob.go
package bob

import (
	"strings"
	"unicode"
)

func Hey(remark string) string {
	remark = strings.TrimSpace(remark)

	if remark == "" {
		return "Fine. Be that way!"
	}

	isQuestion := strings.HasSuffix(remark, "?")
	hasLetters := strings.IndexFunc(remark, unicode.IsLetter) >= 0
	isShout := hasLetters && strings.ToUpper(remark) == remark

	if isShout && isQuestion {
		return "Calm down, I know what I'm doing!"
	}
	if isShout {
		return "Whoa, chill out!"
	}
	if isQuestion {
		return "Sure."
	}
	return "Whatever."
}
